lib/teams/server/message-actions/chat-message-actions.js

```diff
@@ -0,0 +1,235 @@
+                TeamsConst.CHAT.MESSAGE.INVITE_ACCEPTED,
+                playerJoiningName
+            );
+            this.sendMessage(message, 'Clan', clan.ownerClient);
+        });
+    }
+
+    inviteClanRejectedEventListener()
+    {
+        this.events.on('reldens.clanJoinInviteRejected', (props) => {
+            let clientSendingInvite = sc.get(props, 'clientSendingInvite', false);
+            let playerRejectingName = sc.get(props, 'playerRejectingName', false);
+            if (!clientSendingInvite ||
+                !playerRejectingName) {
+                return false;
+            }
+            let message = this.createMessage(
+                TeamsConst.CHAT.MESSAGE.INVITE_REJECTED,
+                playerRejectingName
+            );
+            this.sendMessage(message, 'Clan', clientSendingInvite);
+        });
+    }
+
+    clanMemberLeavingEventListener()
+    {
+        this.events.on('reldens.clanLeaveBeforeSendUpdate', (props) => {
+            let playerLeavingId = sc.get(props, 'playerId', false);
+            if (!playerLeavingId) {
+                Logger.error('playerLeavingId undefined on ChatMessageAction');
+                return false;
+            }
+            let currentClan = sc.get(props, 'currentClan', false);
+            if (!currentClan) {
+                Logger.error('currentClan undefined on ChatMessageAction');
+                return false;
+            }
+            let disbandClan = sc.get(props, 'disbandClan', false);
+
+            if (disbandClan && playerLeavingId !== currentClan.owner.id) {
+                this.clanDisbanding(currentClan.owner.playerName, currentClan.clients[playerLeavingId]);
+                return true;
+            }
+            let message = this.createMessage(TeamsConst.CHAT.MESSAGE.LEAVE,
+                currentClan.players[playerLeavingId].playerName,
+                'clan');
+            this.sendMessage(message, 'Clan', currentClan.ownerClient);
+        });
+    }
+
+    clanDisbanding(clanOwnerName, clientLeaving)
+    {
+        let message = this.createMessage(TeamsConst.CHAT.MESSAGE.DISBANDED,
+            clanOwnerName,
+            'clan');
+        this.sendMessage(message, 'Clan', clientLeaving);
+    }
+
+    createMessage(baseChatMessage, playerName, groupName = false)
+    {
+        let message = baseChatMessage.replace('%playerName', playerName);
+        if (groupName) {
+            message = message.replace('%groupName', groupName);
+        }
+        return message;
+    }
+
+    sendMessage(message, chatFrom, client)
+    {
+        let messageObject = {
+            act: ChatConst.CHAT_ACTION,
+            [ChatConst.CHAT_FROM]: chatFrom,
+            [ChatConst.CHAT_MESSAGE]: message,
+            [ChatConst.CHAT_TO]: ChatConst.CHAT_TYPE_SYSTEM
+        };
+        client.send('*', messageObject);
+
+        // this.chatPlugin.chatManager.saveMessage();
+    }
+
+    gatherOtherPlayersOnTeam(team, filterIds)
```

Comment:
over complex function, no need to check on isFiltered, neither pass the "filtered" ones, check how I did it on the TeamUpdatesHandler, you can "clone" the entire playersList everytime and just remove the current looped one